lib/r10k/environment/with_modules.rb

```diff
@@ -101,10 +103,22 @@ def validate_no_module_conflicts
                 .select { |_, v| v.size > 1 }
                 .map(&:first)
     unless conflicts.empty?
-      msg = _('Puppetfile cannot contain module names defined by environment %{name}') % {name: self.name}
-      msg += ' '
-      msg += _("Remove the conflicting definitions of the following modules: %{conflicts}" % { conflicts: conflicts.join(' ') })
-      raise R10K::Error.new(msg)
+      conflicts_str = conflicts.join(', ')
+      log_msg = _('Environment and Puppetfile both define the following modules, Puppetfile ' \
+                  'definition will be ignored: %{mods}' % { mods: conflicts_str })
+      case conflict_opt = @options[:module_conflicts]
+      when 'override_puppetfile'
+        logger.debug log_msg
+      when 'override_puppetfile_and_warn', nil
```

Comment:
I didn't notice this before, but what's the `nil` here for?

Reply:
If the setting isn't specified, the option is nil. Default behavior. I don't _think_ r10k has a nice way of using schemas or other systems for this kind of thing? Could be wrong. Any suggestions?

Reply:
ahhhh, gotcha. I think it's fine, I'm just a little out of practice at reading ruby haha 😅 
I had skimmed and thought that the first case was the default, so wasn't sure what this was doing here in the middle, but I see now.